packages/flutter/lib/src/rendering/box.dart

```diff
@@ -2362,14 +2364,30 @@ abstract class RenderBox extends RenderObject {
       _cachedBaselines?.clear();
       _cachedIntrinsicDimensions?.clear();
       _cachedDryLayoutSizes?.clear();
-      if (parent is RenderObject) {
-        markParentNeedsLayout();
-        return;
-      }
+      _notifyParentIfDirty = true;
+    }
+  }
+
+  @override
+  void markNeedsLayout() {
+    _clearCachedData();
+    if (_notifyParentIfDirty && parent is RenderObject) {
+      markParentNeedsLayout();
+      _notifyParentIfDirty = false;
+      return;
     }
     super.markNeedsLayout();
   }
 
+  @override
+  void layout(Constraints constraints, {bool parentUsesSize = false}) {
+    if (constraints != _lastConstraints) {
+      _clearCachedData();
```

Comment:
I think we only need to invalidate the baseline cache here? The intrinsics calculations are per definition independent of constraints.

Reply:
1) Clearing the cache unnecessarily will make it less useful.
2) Per definition, intrinsics are independent of constraints. If an RO uses constraints to calculate intrinsics, that's a bug. It will also be hard to write such an implementation since the intrinsics methods are usually called before layout...
3) `_cachedDryLayoutSizes` is dependent on the constraints that were passed to `computeDryLayout` (not the constraints passed to layout here). Furthermore, the dry layout size cache is keyed on constraints - and for the same constraints it should always calculate the same intrinsics size (all things being equal). So, clearing the cache when constraints change is not necessary.

Reply:
> 1) Clearing the cache unnecessarily will make it less useful.
> 
> 2) Per definition, intrinsics are independent of constraints. If an RO uses constraints to calculate intrinsics, that's a bug. It will also be hard to write such an implementation since the intrinsics methods are usually called before layout...
> 
> 3) `_cachedDryLayoutSizes` is dependent on the constraints that were passed to `computeDryLayout` (not the constraints passed to layout here). Furthermore, the dry layout size cache is keyed on constraints - and for the same constraints it should always calculate the same intrinsics size (all things being equal). So, clearing the cache when constraints change is not necessary.

Yes I understand now. Thanks for explaining this.